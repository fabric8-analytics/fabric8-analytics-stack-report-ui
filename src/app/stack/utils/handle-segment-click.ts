import { TokenDetailModel } from '../models/stack-report.model';

export class HandleSegmentClick {

    /**
     * trackCardClick
     */
    public trackCardClick(e: MouseEvent, tokenDetail: TokenDetailModel, title: string) {

        var userId = tokenDetail.id;
        var registrationStatus = tokenDetail.status;

        if (e.isTrusted) {
            analytics.identify(userId, {
                registrationStatus: registrationStatus,
                userClickedOn: title

            })
        }

    }
}