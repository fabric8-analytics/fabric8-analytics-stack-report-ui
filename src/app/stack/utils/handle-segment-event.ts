import { TokenDetailModel } from "../models/stack-report.model";

/**
 * identifyUser
 */
export function identifyUser(userId: string, registrationStatus: string) {
    analytics.identify(userId, {
        registrationStatus: registrationStatus,
    });
}

export function trackClick(e: MouseEvent, title: string, location?: string) {
    if (e.isTrusted) {
        if (location) {
            analytics.track(title, {
                location: location,
            });
        } else {
            analytics.track(title);
        }
    }
}
