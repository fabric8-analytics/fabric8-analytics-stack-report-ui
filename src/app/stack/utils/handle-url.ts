export class HandleUrl {
    public open(event, url = null) {
        event.preventDefault();
        event.stopPropagation();
        let elem: HTMLElement = <HTMLElement>event.target;
        if (elem && elem.attributes && elem.attributes.getNamedItem("href")) {
            let url = event.target.href;
            parent.postMessage(url, "*");
        } else if (url) {
            parent.postMessage(url, "*");
        }
    }
}
