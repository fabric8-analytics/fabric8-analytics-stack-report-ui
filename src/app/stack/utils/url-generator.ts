export class GenerateUrl {

    public regitrationURL: string = "https://app.snyk.io/signup/?utm_medium=Partner&utm_source=Red-Hat&utm_campaign=Code-Ready-Analytics-2020&utm_content=register";

    public publicUrl(url: string): string {
        if (typeof (url) === "string") {
            let content: Array<string>;
            let generatedUrl: string;
            content = url.split('/vuln/', 2);
            generatedUrl = url + "?utm_medium=Partner&utm_source=Red%20Hat&utm_campaign=Code-Ready-Analytics-2020&utm_content=vuln/" + content[1];
            return generatedUrl;
        }
        return null;
    }
    public privateUrl(url: string, regitrationStatus: string): string {
        if (regitrationStatus === 'freetier') {
            return this.regitrationURL;
        }
        else if (typeof (url) === "string") {
            let content: Array<string>;
            let generatedUrl: string;
            content = url.split('/vuln/', 2);
            generatedUrl = url + "?utm_medium=Partner&utm_source=Red%20Hat&utm_campaign=Code-Ready-Analytics-2020&utm_content=vuln/" + content[1];
            return generatedUrl;
        }
        return null;
    }
}