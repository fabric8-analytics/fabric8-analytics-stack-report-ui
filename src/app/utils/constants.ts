const stageApiUrl =
  "https://f8a-analytics-preview-2445582058137.staging.gw.apicast.io";
const stageApiKey = "3e42fa66f65124e6b1266a23431e3d08";

const prodApiUrl =
  "https://f8a-analytics-2445582058137.production.gw.apicast.io";
const prodApiKey = "9e7da76708fe374d8c10fa752e72989f";

const localDevURL = "http://localhost:8080/";

export { prodApiUrl, prodApiKey, stageApiUrl, stageApiKey, localDevURL };

//  https://f8a-analytics-preview-2445582058137.staging.gw.apicast.io/api/v2/stack-analyses/ed16d5db91d64e8d8d0ea6d1ba6b727e?user_key=3e42fa66f65124e6b1266a23431e3d08
