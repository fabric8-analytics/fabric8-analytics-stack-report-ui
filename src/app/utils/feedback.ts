/* eslint-disable no-console */
import axios, { AxiosResponse, AxiosError } from "axios";
import { Logger } from "./logger";

/* const submitFeedback = (data: any, feedback: string) => {
  const urlf = "https://recommender.api.openshift.io/api/v1/user-feedback";
  axios({
    method: "post",
    url: urlf,
    headers: {},
    data: {
      foo: "bar", // This is the body part
    },
  }).then((response) => {
    console.log(response);
  });
};



export default submitFeedback;

*/
