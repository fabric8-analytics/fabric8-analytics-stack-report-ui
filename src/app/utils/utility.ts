export default class Utils {

    GetTransitiveCountHelper = (element: any): number => {
        let TransitiveCount = 0;
        console.log(element)
        if(element.dependencies != null && element.dependencies.length > 0){
            TransitiveCount += element.dependencies.length
            return TransitiveCount

        }
        return 0;
    }

    // @ts-ignore 
    GetTotalVulnerabilityCount = (data: any) => {
        let TotalVulnerabilities = 0;
        let VulnerableDeps = 0;
        let c = 0
        if(data?.length !== 0){

            data?.forEach((element: any)=> {
                c += 1
                const publicVulns = element.private_vulnerabilities.length;
                const privateVulns = element.public_vulnerabilities.length;
                const vulnerable = element.vulnerable_dependencies.length;
                // const count = publicVulns + privateVulns
            
                if(publicVulns > 0 || privateVulns > 0 || vulnerable > 0){
                    const countVulnerable = publicVulns + privateVulns + vulnerable
                    const count = publicVulns + privateVulns
                    TotalVulnerabilities += count
                    VulnerableDeps += countVulnerable + this.VulnerableHelper(element.vulnerable_dependencies);
                }
                
                
            });
            
            return {"total": TotalVulnerabilities, "vulnerable": VulnerableDeps}
        }

        return {"total": 0, "vulnerable": 0};
    }

    VulnerableHelper = (data: any): number => {
        let Count = 0
        data?.forEach((element: any) => {
            if(element.vulnerable_dependencies != null && element.vulnerable_dependencies.length > 0){
                Count += element.vulnerable_dependencies.length + this.VulnerableHelper(element);
            }

        });
        return Count;
    }


    GetTransitiveCount = (data: any): number => {
        let TransitiveCount = 0;
        if (data?.length !== 0){
            data?.forEach((deps: any) => {
                const element = deps?.dependencies
                if(element != null && element.length > 0){
                    TransitiveCount += element.length + this.GetTransitiveCount(element)

                    console.log(TransitiveCount);
                    
                }
                
            });
            return TransitiveCount;
            
        }
        return 0;
        
    }
}