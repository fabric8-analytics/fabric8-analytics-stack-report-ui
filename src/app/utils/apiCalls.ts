import axios, { AxiosResponse, AxiosError } from "axios";
import { useContext, useEffect } from "react";
import Context from "../store/context";
import { stageApiKey, stageApiUrl, localDevURL } from "./constants";
import { Logger } from "./logger";

function GetStackDetails(requestId = "ed16d5db91d64e8d8d0ea6d1ba6b727e") {
  // @ts-ignore
  const { globalState, globalDispatch } = useContext(Context);
  useEffect(() => {
    globalDispatch({ type: "Loading", data: true });
    const url = `${stageApiUrl}/api/v2/stack-analyses/${requestId}?user_key=${stageApiKey}`;
    Logger.log(url);
    const config = {
      headers: {
        UUID: "2527bcb9-9db1-4c8f-b304-c3b232ebd22f",
      },
    };
    Logger.log(config);
    const devUrl = localDevURL;
    axios
      .get(url)
      .then((res: { data: any }) => {
        globalDispatch({ type: "APIData", data: res.data });
        globalDispatch({ type: "Loading", data: false });
        Logger.log(res.data);
      })
      .catch((err) => {
        Logger.log(err);
        Logger.log(globalState);
        Logger.log(globalDispatch);
      });
  }, []);
}

function RegisterUser(snyk_token = "") {
  // @ts-ignore
  const { globalState, globalDispatch } = useContext(Context);

  useEffect(() => {
    const UUID = localStorage.getItem("UUID");
    if (UUID === null) {
      const url = `${stageApiUrl}/user?user_key=${stageApiKey}`;
      axios
        .post(url)
        .then((res: AxiosResponse) => {
          const result = res.data;
          // @ts-ignore
          if (res.status === 200) {
            const data = {
              snyk_api_token: snyk_token,
              user_id: result.user_id,
            };
            axios
              .put(url, data)
              .then((response: AxiosResponse) => {
                if (response.status === 200) {
                  globalDispatch({ type: "UUID", data: response.data.user_id });
                  localStorage.setItem("UUID", response.data.user_id);
                }
              })
              .catch((error) => {
                if (error.response) {
                  Logger.log(error.response.data);
                  Logger.log(error.response.status);
                }
              });
          }
        })
        .catch((error) => {
          if (error.response) {
            Logger.log(error.response.data);
            Logger.log(error.response.status);
          }
        });
    }
  }, []);
}

export { GetStackDetails, RegisterUser };
