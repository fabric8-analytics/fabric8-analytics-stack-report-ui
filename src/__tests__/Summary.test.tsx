import React from "react";
import { render } from "@testing-library/react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import SummaryCard from "../app/components/main-components/Summary/Summary";
import { GetStackDetails } from "../app/utils/apiCalls";
import Licenses from "../app/components/shared-components/licenses-detail/licenses-detail";

describe("Summary Card components Unit Tests ", () => {
  test("Should be able to render the Summary Card component", async () => {
    GetStackDetails("5264cdab4838457d9c7ce2a420039dd2", "valid-uuid");
    const component = shallow(<SummaryCard />);
    expect(component).toMatchSnapshot();
  });
  test("I Should be able to render the licenses component", async () => {
    const MocLicenseProps = { conflicts: 7, unknown: 8 };
    const component = shallow(<Licenses conflicts={0} unknown={0} />);
    component.setProps(MocLicenseProps);
    // expect(component.prop("unknown")).toBeDefined();
    expect(toJson(component)).toMatchSnapshot();
    expect(toJson(component.find("#unknown").children())).toBe(
      MocLicenseProps.unknown,
    );
    expect(toJson(component.find("#confilcts").children())).toBe(
      MocLicenseProps.conflicts,
    );
  });
});
