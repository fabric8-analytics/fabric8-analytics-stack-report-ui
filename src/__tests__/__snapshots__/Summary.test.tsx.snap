// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Card components Unit Tests  I Should be able to render the licenses component 1`] = `
<div>
  <Flex
    direction={
      Object {
        "default": "column",
      }
    }
    display={
      Object {
        "default": "inlineFlex",
      }
    }
    key="1"
  >
    <FlexItem
      spacer={
        Object {
          "default": "spacerXs",
        }
      }
    >
      <Split
        hasGutter={true}
      >
        <SplitItem>
          <FileAltIcon
            className="icon-class"
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </SplitItem>
        <SplitItem>
          <Title
            headingLevel="h6"
            size="md"
          >
            Licences
          </Title>
        </SplitItem>
        <SplitItem>
          <ExclamationCircleIcon
            className="icon-class-exclamation"
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </SplitItem>
      </Split>
    </FlexItem>
  </Flex>
  <Flex
    direction={
      Object {
        "default": "column",
      }
    }
    key="2"
  >
    <FlexItem
      spacer={
        Object {
          "default": "spacerXs",
        }
      }
    >
      <Split
        hasGutter={true}
      >
        <SplitItem>
          License conflicts:
        </SplitItem>
        <SplitItem>
          <Title
            headingLevel="h6"
            id="confilcts"
            size="md"
          >
            7
          </Title>
        </SplitItem>
      </Split>
    </FlexItem>
    <FlexItem
      spacer={
        Object {
          "default": "spacerXs",
        }
      }
    >
      <Split
        hasGutter={true}
      >
        <SplitItem>
          Unknown licenses:
        </SplitItem>
        <SplitItem>
          <Title
            headingLevel="h6"
            id="unknown"
            size="md"
          >
            8
          </Title>
        </SplitItem>
      </Split>
    </FlexItem>
  </Flex>
</div>
`;

exports[`Summary Card components Unit Tests  Should be able to render the Summary Card component 1`] = `
<Card
  className="GridCard"
  isFullHeight={true}
  isHoverable={true}
>
  <CardTitle
    className=""
  >
    <Flex>
      <FlexItem>
        <TextContent>
          <Text
            className="heading"
          >
             Summary of the stack 
          </Text>
        </TextContent>
      </FlexItem>
      <FlexItem>
        <BTSynktoken
          isUUID={false}
        />
      </FlexItem>
      <FlexItem
        align={
          Object {
            "default": "alignRight",
          }
        }
      >
        <ShareSquareIcon
          className="icon-class"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </FlexItem>
      <FlexItem>
        <DownloadIcon
          className="icon-class"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </FlexItem>
    </Flex>
  </CardTitle>
  <Divider />
  <CardBody>
    <PoweredBySynk />
    <SignUp
      isUUID={false}
    />
    <br />
    <Grid
      hasGutter={true}
      span={12}
      xl={6}
      xl2={6}
    >
      <GridItem
        lg={6}
        md={6}
        sm={12}
      >
        <Dependency
          analyzed={0}
          transitive={0}
          unknown={0}
        />
      </GridItem>
      <GridItem
        lg={6}
        md={6}
        sm={12}
      >
        <Security
          vulnerable={0}
          vulnerablities={0}
        />
      </GridItem>
    </Grid>
    <br />
    <Divider
      component="div"
    />
    <br />
    <Grid
      hasGutter={true}
      span={12}
      xl={6}
      xl2={6}
    >
      <GridItem
        lg={6}
        md={6}
        rowSpan={4}
        sm={12}
      >
        <Licenses
          conflicts={0}
          unknown={0}
        />
      </GridItem>
      <GridItem
        lg={6}
        md={6}
        rowSpan={4}
        sm={12}
      >
        <Addons
          companion={0}
        />
      </GridItem>
    </Grid>
  </CardBody>
  <Divider
    component="div"
  />
</Card>
`;
